sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfMetadataSequencer.java

```diff
@@ -0,0 +1,238 @@
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.CREATE_DATE, metadata.getCreateDate());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.CREATOR_TOOL, metadata.getCreatorTool());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.IDENTIFIER, metadata.getIdentifier());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.METADATA_DATE, metadata.getMetadataDate());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.MODIFY_DATE, metadata.getModifyDate());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.NICKNAME, metadata.getNickname());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.RATING, metadata.getRating());
+                setPropertyIfMetadataPresent(xmpNode, XmpMetadataLexicon.LABEL, metadata.getLabel());
+                return true;
+            }
+        } catch (Exception e) {
+            getLogger().error(e, "Couldn't process stream.");
+        }
+
+        return false;
+    }
+
+    private Node getPdfMetadataNode( Node outputNode ) throws RepositoryException {
+        if (outputNode.isNew()) {
+            outputNode.setPrimaryType(PdfMetadataLexicon.METADATA_NODE);
+            return outputNode;
+        }
+        return outputNode.addNode(PdfMetadataLexicon.METADATA_NODE, PdfMetadataLexicon.METADATA_NODE);
+    }
+
+    private void setPropertyIfMetadataPresent( Node node,
+                                               String propertyName,
+                                               Object value ) throws RepositoryException {
+        if (value != null) {
+            if (value instanceof String && !StringUtil.isBlank((String) value)) {
+                node.setProperty(propertyName, (String) value);
+            } else if (value instanceof Boolean) {
+                node.setProperty(propertyName, (Boolean) value);
+            } else if (value instanceof Long) {
+                node.setProperty(propertyName, (Long) value);
+            } else if (value instanceof Integer) {
+                node.setProperty(propertyName, new Long((Integer) value));
+            } else if (value instanceof Calendar) {
+                // pdfbox 1.8.x doesn't parse the timezones correctly...
+                // see PDFBOX-3352
+                Calendar calendarValue = (Calendar) value;
+                if (calendarValue.getTimeZone().getID().toLowerCase().equals("unknown")) {
+                    calendarValue.setTimeZone(TimeZone.getDefault());
+                }
+                node.setProperty(propertyName, calendarValue);
+            } else if (value instanceof byte[]) {
+                InputStream is = new ByteArrayInputStream((byte []) value);
+                Binary binaryProperty = node.getSession().getValueFactory().createBinary(is);
+                node.setProperty(propertyName, binaryProperty);
+            } else if (value instanceof List) {
```

Comment:
if this uses unchecked types, should there be an `@SupressWarnings("unchecked")` in this method ?
Either that or change to `List<?>`